test/GLMTest.m

```diff
@@ -280,4 +453,135 @@ function test_cortical_simulation(testCase)
+    events(2).times(2,i_event)=t+duration_stim1+duration_stim2;
+   
+    t=t+duration_stim1+duration_stim2;
+    i_event=i_event+1;
+    
+end
+
+events(1).epochs = ones(1, length(events(1).times));
+events(2).epochs = ones(1, length(events(2).times));
+
+hrf_types = process_nst_glm_fit('get_hrf_types');
+[X,names] = process_nst_glm_fit('make_design_matrix', time, events, ...
+                                hrf_types.CANONICAL, 25, 1);
+
+                            
+assert(all(strcmp(names(1:2), {events.label})));
+
+end
+
+function [B, active_channels]= create_B_map(nb_channels,nb_active_channels)
+    % Assume 2 conditions Stim1=Task,Stim2=Rest  
+    B=zeros(3,nb_channels);
+    perm=randperm(nb_channels);
+    
+    active_channels=sort(perm(1:nb_active_channels));
+    unactive_channels=sort(perm(nb_active_channels+1:end));
+    
+    
+    % Active channels : B1 = 0.01; B2~ 0.005, B3=N(0,0.0001)
+    B(1,active_channels)=0.01+0.0001*randn(); % Stim 1
+    B(2,active_channels)=0.005+0.0001*randn(); % Stim 2 
+    B(3,active_channels)=0.0001*randn(); % Constant 
+
+    % Active channels : B1 =  N(0,0.001); B2~ N(0,0.001), B3=0
+
+    B(1,unactive_channels)=0.001*randn();% Stim 1
+    B(2,unactive_channels)=0.001*randn(); % Stim 2 
+    B(3,unactive_channels)=0.0001*randn(); % Constant 
+    
+    B=B/100;
+end
+
+
+function l=get_factor_SNR(signal,noise,SNR)
+    % compute l so that y = signal + l*noise avec the correct SNR value
+    % SNR = 10 log(  <signal> / l^2 < noise > ) with <.> the signal power.
+    
+    mss_signal=norm(signal,2);
+    mss_noise=norm(noise,2);
+    
+    l2= (mss_signal/mss_noise) * 10^(-SNR/10);
+    l=sqrt(l2);
+end
+
+function [TP,FP,TN,FN]=compute_performance(glm_results, true_results)
+    FP= sum(glm_results( true_results == 0  ));
+    TP= sum(glm_results( true_results >  0 ));
+    
+    TN= sum(~glm_results(true_results == 0));
+    FN= sum(~glm_results(true_results  >  0 ));
+end
+
+function plot_performance(glm_results_stat,true_map)
+
+    pval= logspace(-7,0,1000);
+    
+    N=length(pval);
+    
+    FPR=zeros(1,N);
+    sensitivity=zeros(1,N);
+
+    for i=1:N
+        thresh_options.pThreshold = pval(i);
+        thresh_options.Correction = 'bonferroni';
+        thresh_options.Control = 1;
+
+        [glm_results, corr_p] = bst_stat_thresh(glm_results_stat.pmap, thresh_options);
+        [TP,FP,TN,FN]=compute_performance(glm_results, true_map);
+
+        FPR(i)=FP/(FN+TP);
+        sensitivity(i)=TP/(TP+FN);
+        
+    end
+    
+    figure(1)
+    plot(FPR,sensitivity);
+    xlabel('False Positive Rate')
+    ylabel('Sensitivity')
+    
+    figure(2)
+    plot(pval,FPR);
+    xlabel('p-value (Bonferroni correction)')
+    ylabel('False Positive Rate')
+    
+    figure(3)
+    plot(pval,sensitivity);
+    xlabel('p-value (Bonferroni correction)')
+    ylabel('Sensitivity')
+
+    save( ['/NAS/home/edelaire/Documents/output_analysis/simulation/GLM_whit_filterd_FPR.mat'], 'FPR');
```

Comment:
Should be removed (specific to your setup)
It's better to do assertion and checks directly in the unit test;

Maybe it's data you want to save because you need it somewhere else. Then it's better to use a dedicated script (and maybe call a factorized function used in the unit test)